### PR TITLE
SQL: Fix issue with getting DATE type in JDBC

### PR DESCRIPTION
**Pull request by matriv**
Previously, calling getDate()/getTime()/getTimestamp() and getObject()
with the corresponding java.sql class on a column of SQL DATE type from
the JDBC result set would throw an Exception.

### CODE REVIEW

**Comment by matriv** [comment]
Needs #40163 to be merged first for tests to pass.

File: x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSet.java [hunk 1]
```diff
@@ -266,6 +266,10 @@ private Long dateTime(int columnIndex) throws SQLException {
         }
     }
 
+    private EsType getEsType(int columnIndex) {
```
Comment (costin) [inline x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSet.java]: I think `columnType` is more appropriate (see `columns` or `column`).

**Review by costin — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM

**Comment by matriv** [comment]
@elasticmachine run elasticsearch-ci/default-distro

**Comment by matriv** [comment]
@elasticmachine run elasticsearch-ci/default-distro

**Comment by matriv** [comment]
@elasticmachine run elasticsearch-ci/default-distro

**Comment by matriv** [comment]
Backported to `7.x` with e1eb683c51ff8bcd09613543e7faf1cdfae9d509
to `7.0` with c3fb2be270910a71bb21263e388f8c2d3d3e0e83
to `6.7` with e9ab0879077df9f07c783005a4da0b37cd8a1d90

**Pull request merged**